File: accudrop/app/src/main/java/me/chrislane/accudrop/viewmodel/ReplayViewModel.java
package me.chrislane.accudrop.viewmodel;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;
import android.location.Location;
import android.support.v4.util.Pair;

import java.util.List;
import java.util.UUID;

public class ReplayViewModel extends ViewModel {

    private MutableLiveData<Integer> jumpId = new MutableLiveData<>();
    private MutableLiveData<Integer> firstJumpId = new MutableLiveData<>();
    private MutableLiveData<Integer> lastJumpId = new MutableLiveData<>();
    private MutableLiveData<List<Pair<UUID, List<Location>>>> usersAndLocs = new MutableLiveData<>();

    /**
     * Get the jump ID for the replay.
     *
     * @return The jump ID for the replay.
     */
    public LiveData<Integer> getJumpId() {
        return jumpId;
    }

    /**
     * Set the jump ID for the replay.
     *
     * @param jumpId The jump ID for the replay.
     */
    public void setJumpId(int jumpId) {
        this.jumpId.setValue(jumpId);
    }

    public LiveData<List<Pair<UUID, List<Location>>>> getUsersAndLocs() {
        return usersAndLocs;
    }

    public void setUsersAndLocs(List<Pair<UUID, List<Location>>> usersAndLocs) {
        this.usersAndLocs.setValue(usersAndLocs);
    }

    public void setFirstJumpId(int firstJumpId) {
        this.firstJumpId.setValue(firstJumpId);
    }

    public void setLastJumpId(int lastJumpId) {
        this.lastJumpId.setValue(lastJumpId);
    }

    public LiveData<Integer> getLastJumpId() {
        return lastJumpId;
    }

    public MutableLiveData<Integer> getFirstJumpId() {
        return firstJumpId;
    }
}

File: app/src/main/java/me/chrislane/accudrop/viewmodel/ReplayViewModel.kt
package me.chrislane.accudrop.viewmodel

import android.location.Location
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import java.util.*

class ReplayViewModel : ViewModel() {

    private val jumpId = MutableLiveData<Int>()
    private val firstJumpId = MutableLiveData<Int>()
    private val lastJumpId = MutableLiveData<Int>()
    private val usersAndLocs = MutableLiveData<MutableList<Pair<UUID, MutableList<Location>>>>()

    /**
     * Get the jump ID for the replay.
     *
     * @return The jump ID for the replay.
     */
    fun getJumpId(): LiveData<Int> {
        return jumpId
    }

    /**
     * Set the jump ID for the replay.
     *
     * @param jumpId The jump ID for the replay.
     */
    fun setJumpId(jumpId: Int) {
        this.jumpId.value = jumpId
    }

    fun getUsersAndLocs(): LiveData<MutableList<Pair<UUID, MutableList<Location>>>> {
        return usersAndLocs
    }

    fun setUsersAndLocs(usersAndLocs: MutableList<Pair<UUID, MutableList<Location>>>) {
        this.usersAndLocs.value = usersAndLocs
    }

    fun setFirstJumpId(firstJumpId: Int) {
        this.firstJumpId.value = firstJumpId
    }

    fun setLastJumpId(lastJumpId: Int) {
        this.lastJumpId.value = lastJumpId
    }

    fun getLastJumpId(): LiveData<Int> {
        return lastJumpId
    }

    fun getFirstJumpId(): LiveData<Int> {
        return firstJumpId
    }
}
